private/buf/buffetch/internal/reader.go

```diff
@@ -379,30 +524,50 @@ func (r *reader) getGitBucket(
 	); err != nil {
 		return nil, fmt.Errorf("could not clone %s: %v", gitURL, err)
 	}
-	terminateFileDirectoryPath, err := findTerminateFileDirectoryPathFromBucket(ctx, readWriteBucket, subDirPath, terminateFileNames)
+	terminateFileProvider, err := findTerminateFileDirectoryPathFromBucket(ctx, readWriteBucket, subDirPath, terminateFileNames)
 	if err != nil {
 		return nil, err
 	}
+	terminateFileDirectoryPath := ""
+	// Get the highest priority file found and use it as the terminate file directory path.
+	terminateFiles := terminateFileProvider.GetTerminateFiles()
+	if len(terminateFiles) != 0 && terminateFiles[0] != nil {
+		terminateFileDirectoryPath = terminateFiles[0].Path()
+	}
```

Comment:
We're performing this same logic several times - should we expose a `GetHighestPriority` method on the `TerminateFileProvider` so that we can re-use it elsewhere?

Reply:
Yeah, I think that is reasonable, since it is encapsulated by the `TerminateFileProvider` and is used for all the different refs :D

Reply:
Now that I think about this more, I guess it would really just be a helper to resolve the first file in the slice.. and that wouldn't be very helpful. Indexing into the first element seems fine as-is.

Reply:
Lol sounds good -- yeah, I implemented the change and it turned out to be the same LOC (+ the extra LOC for defining the helper itself and extending the interface, lol). I can try and take a generally more critical approach around DRY-ing up this file.